### PR TITLE
add profile tests (non-legacy)

### PR DESCRIPTION
**Pull request by arlimus**
we missed these last time around when rspec and our tests were clashing. fixed and updated.

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @arlimus

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: lib/inspec/targets/dir.rb [hunk 1]
```diff
@@ -11,13 +11,11 @@ module DirsHelper
     # TODO: remove `test` support for InSpec 1.0
     class ProfileDir
       def handles?(paths)
+        return true if paths.include?('inspec.yml')
```
Comment (arlimus) [inline lib/inspec/targets/dir.rb:14]: i.e. we don't need `controls` or `test` folder, just finding `inspec.yml` is enough.

Comment (chris-rock) [inline lib/inspec/targets/dir.rb:14]: :+1:

**Comment by chris-rock** [comment]
Thanks @arlimus for this improvement

**Pull request merged**